dome.gd

```diff
@@ -16,9 +16,13 @@ var connections: Dictionary = {}
 
 enum InfestationStage {UNINFESTED, MINOR, MODERATE, MAJOR, FULL, LOST}
 
+func _ready():
+	infestation_rate = Globals.base_infestation_rate
+
 func _process(delta):
 	# Process infestation progression inependently in dome's infestation check
-	add_infestation((infestation_rate + infestation_modifier) * delta)
+	if infestation_percentage > 0:
```

Comment:
I think this is a cleaner way to model infestation growth

Reply:
I like this!